avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -619,82 +676,121 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
-    // struct converts to one of those (record), so for everything else that has a name we store
-    // the full name into a special property. For uniformity, we also duplicate this info into
-    // the same field in records as well even though it will also be available in the namespace()
-    // and name().
-    if (schema.name() != null) {
-      baseSchema.addProp(CONNECT_NAME_PROP, schema.name());
-
-      if(Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())){
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DECIMAL);
-      } else if(Time.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIME_MILLIS);
-      } else if(Timestamp.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIMESTAMP_MILLIS);
-      } else if(Date.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DATE);
+        if (schema.name() != null) {
+          baseSchema.addProp(CONNECT_NAME_PROP, schema.name());
+        }
+        // Some Connect types need special annotations to preserve the types accurate due to
+        // limitations in Avro. These types get an extra annotation with their Connect type
+        if (connectType != null) {
+          baseSchema.addProp(CONNECT_TYPE_PROP, connectType);
+        }
+      }
+      
+      // Only Avro named types (record, enum, fixed) may contain namespace + name. Only Connect's
+      // struct converts to one of those (record), so for everything else that has a name we store
+      // the full name into a special property. For uniformity, we also duplicate this info into
+      // the same field in records as well even though it will also be available in the namespace()
+      // and name().
+      if (schema.name() != null) {
+        if(Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())){
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DECIMAL);
+        } else if(Time.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIME_MILLIS);
+        } else if(Timestamp.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIMESTAMP_MILLIS);
+        } else if(Date.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DATE);
+        }
       }
-    }
 
-    // Some Connect types need special annotations to preserve the types accurate due to
-    // limitations in Avro. These types get an extra annotation with their Connect type
-    if (connectType != null) {
-      baseSchema.addProp(CONNECT_TYPE_PROP, connectType);
-    }
+      if (schema.parameters() != null){
+        for(Map.Entry<String, String> entry : schema.parameters().entrySet())
+        {
+          if (entry.getKey().startsWith(AVRO_PROP)){
```

Comment:
Where are the details of these `avro` prefixed parameters specified? I don't think that's in the Avro spec, so I'm not sure it's a constraint we should apply here. I know `avro.java.string` is probably the most common example, but should we just be copying all parameters? Or is the `avro` prefix actually specified somewhere?